src/utils/formData.ts

```diff
@@ -0,0 +1,30 @@
+import { FieldsObject } from "../Forms/Validation";
+
+const fieldValue = (element: Element) => {
+  const { type, checked, value, id } = element as HTMLInputElement;
+
+  switch (type) {
+    case "checkbox":
+      return { [id]: checked };
+    case "button":
+    case "submit":
+    case "reset":
+      return {};
+    default:
+      return { [id]: value };
+  }
+};
+
+export default function formData(
+  elements?: HTMLFormElement | HTMLFormControlsCollection
```

Comment:
why `element` is optional ? 🤔

Reply:
it used to accept an empty object as a fall back in some usages but to make the code and typing easier I just made it optional